test/CheckoutSdkTest/JsonSerializerTest.cs

```diff
@@ -112,5 +113,47 @@ public void ShouldDeserializeDefaultCardCTypeResponse()
             cardDetailsResponse.ShouldNotBeNull();
             cardDetailsResponse.Type.ShouldBe(CardType.Physical);
         }
+
+        [Fact]
+        public void ShouldSerializeDateTimeFormatsFromJson()
+        {
+            var fileContent = GetJsonFileContent("./Resources/get_financial_actions_response.json");
+            FinancialActionsQueryResponse financialActionsQueryResponse =
+                (FinancialActionsQueryResponse)new JsonSerializer().Deserialize(fileContent,
+                    typeof(FinancialActionsQueryResponse));
+            financialActionsQueryResponse.ShouldNotBeNull();
+            financialActionsQueryResponse.Data[0].ProcessedOn.ShouldNotBeNull();
+        }
+
+        [Fact]
+        public void ShouldDeserializeDateTimeFormatsFromStrings()
+        {
+            string[] dates =
+            {
+                "{\"processed_on\":\"2021-06-08T12:25:01.000Z\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01Z\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01+00:00\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01+0000\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01+00\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1234567\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1234560\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1234500\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1234000\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1230000\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1200000\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.1000000\"}",
+                "{\"processed_on\":\"2021-06-08T12:25:01.0000000\"}",
+            };
+
+            foreach (var date in dates)
+            {
+                PaymentResponse paymentResponse =
+                    (PaymentResponse)new JsonSerializer().Deserialize(date, typeof(PaymentResponse));
+
+                paymentResponse.ShouldNotBeNull();
+                paymentResponse.ProcessedOn.ShouldNotBeNull();
```

Comment:
We need at least assert that most of the dates have `2021-06-08T12:25:01` to make sure it does not change anything

Reply:
👍🏼